src/Common/src/Internal/Cryptography/BasicSymmetricCipherBCrypt.cs

```diff
@@ -30,9 +30,9 @@ public BasicSymmetricCipherBCrypt(SafeAlgorithmHandle algorithm, CipherMode ciph
 
             _hKey = algorithm.BCryptImportKey(key);
 
-            if (effectiveKeyLength != 0)
+            if (ownsParentHandle)
             {
-                Cng.SetEffectiveKeyLength(_hKey, effectiveKeyLength);
+                _hKey.SetParentHandle(algorithm);
```

Comment:
Ah, I didn't register that the RC2 call to obtain the algorithm handle had it in a using statement.  It clearly made sense when I saw the code... but I didn't internalize it.